File: Medium/451-SortCharactersByFrequency.cpp
/*
   @author
   Akshat Atray
   Leetcode- https://leetcode.com/akshatatray/
   LeetCode Problem 451. Sort Characters By Frequency
   Language C++
   Level Medium
*/

class Solution {
public:
    string frequencySort(string s) {
        unordered_map<char, int> mp;
        for (auto& ch : s) mp[ch]++;
        priority_queue <pair<int, int>> pq;
        for (auto& m : mp) pq.push({m.second, m.first});
        string res="";
        while (!pq.empty()) {
            res += string(pq.top().first, pq.top().second);
            pq.pop();
        }
        return res;
    }
};

File: Medium/451-SortCharactersByFrequency.py
'''
    @author
    Prajjwal Maurya
    Leetcode- https://leetcode.com/pmaurya6jun/
    Leetcode Problem 451. Sort Characters By Frequency
    Language Python
    Level Medium
'''

class Solution:
    def frequencySort(self, s: str) -> str:
        hm = {}
        for char in s:
            if char in hm:
                hm[char] += 1
            else:
                hm[char] = 1
        new = dict(sorted(hm.items(), key = lambda item:item[1], reverse = True))
        temp = ''
        for char, freq in new.items():
            temp += (char*freq)
        return temp